core/domain/commit_commands_domain.py

```diff
@@ -0,0 +1,401 @@
+# coding: utf-8
+#
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain object for commit commands."""
+
+from core.domain import collection_domain
+from core.domain import config_domain
+from core.domain import exp_domain
+from core.domain import story_domain
+import utils
+
+
+def validate_dict_keys(actual_keys, required_keys, optional_keys=None):
+    """Validates that actual keys in a dict contain all the required keys
+    and some/all of the optional keys.
+    """
+
+    missing_keys = [key for key in required_keys if key not in actual_keys]
+
+    if not optional_keys:
+        optional_keys = []
+
+    extra_keys = [key for key in actual_keys if key not in (
+        required_keys + optional_keys)]
+
+    error_msg = ''
+    if len(missing_keys):
+        error_msg = error_msg + 'Following required keys are missing: %s' % (
+            (',').join(missing_keys))
+        if len(extra_keys):
+            error_msg = error_msg + ', '
+
+    if len(extra_keys):
+        error_msg = error_msg + 'Following extra keys are present: %s' % (
+            (',').join(extra_keys))
+
+    if len(missing_keys + extra_keys):
+        raise utils.ValidationError(error_msg)
+
+
+class ExplorationCommitCmd(object):
+    """Domain object for commit commands of a exploration.
+
+    Attributes:
+        name: str. The command name.
+        parameters: dict. The command parameters.
+    """
+
+    def __init__(self, name, parameters):
+        """Constructs a ExplorationCommitCmd domain object.
+
+        Args:
+            name: str. The command name.
+            parameters: dict. The command parameters.
+        """
+        self.name = name
+        self.parameters = parameters
+
+    def validate(self):
+        """Checks that all parameters of a command are valid for the
+        given command name.
+
+        Raises:
+            Exception: The command is invalid.
+        """
+        if self.name == exp_domain.CMD_ADD_STATE or (
```

Comment:
Instead of having multiple if/else, create a command list initially containing commands with required and optional keys and then run a for loop to validate each of the commands (similar to what you have done in tests)

Reply:
Also, add a separate function which iterates over a given command list and validates it instead of repeating the loop each time.

Reply:
Done.